core/components/organisms/textField/__test__/Textarea.test.tsx

```diff
@@ -0,0 +1,222 @@
+  });
+});
+
+describe('TextField Component', () => {
+  test('renders with basic props', () => {
+    const onChangeMock = jest.fn();
+    const onBlurMock = jest.fn();
+    const onFocusMock = jest.fn();
+    const onClearMock = jest.fn();
+
+    render(
+      <TextField
+        label="Test Label"
+        name="test-name"
+        placeholder="Test Placeholder"
+        value="Test Value"
+        defaultValue="Default Value"
+        onChange={onChangeMock}
+        onBlur={onBlurMock}
+        onFocus={onFocusMock}
+        onClear={onClearMock}
+      />
+    );
+
+    // Check if the label is rendered
+    expect(screen.getByText('Test Label')).toBeInTheDocument();
+
+    // Check if the input has the correct props
+    const input = screen.getByPlaceholderText('Test Placeholder') as HTMLInputElement;
+    expect(input).toHaveAttribute('name', 'test-name');
+    expect(input).toHaveValue('Test Value');
+
+    // Trigger events and check if the callbacks are called
+    fireEvent.change(input, { target: { value: 'New Value' } });
+    expect(onChangeMock).toHaveBeenCalled();
+    fireEvent.blur(input);
+    expect(onBlurMock).toHaveBeenCalled();
+    fireEvent.focus(input);
+    expect(onFocusMock).toHaveBeenCalled();
+  });
+
+  test('renders with textarea', () => {
+    render(<TextField withTextarea={true} placeholder="here is the placeholder" />);
+    // Check if the component renders with textarea
+    const textareaElement = screen.getByPlaceholderText('here is the placeholder');
+
+    // Assert that the element is an exact <textarea> instance
+    expect(textareaElement).toBeInTheDocument();
+    expect(textareaElement).toBeInstanceOf(HTMLTextAreaElement);
+    expect(textareaElement.tagName.toLowerCase()).toBe('textarea');
+  });
+
+  test('renders with help text and character count', () => {
+    render(<TextField helpText="Helpful text" max={200} />);
+    // Check if the help text is rendered
+    expect(screen.getByText('Helpful text')).toBeInTheDocument();
+
+    // Check if character count is initially 0/200
+    expect(screen.getByText('200')).toBeInTheDocument();
+  });
+
+  test('input exceeds maximum character count', () => {
+    const onChangeMock = jest.fn();
+    const { getByTestId } = render(
+      <TextField
+        label="Test Label"
+        name="test-name"
+        placeholder="Test Placeholder"
+        value="Test Value"
+        error={true}
+        defaultValue="Default Value"
+        max={298}
+        helpText="here is the helptext"
+        onChange={onChangeMock}
+      />
+    );
+    const textarea = screen.getByPlaceholderText('Test Placeholder') as HTMLTextAreaElement;
+
+    // Type more than the allowed character count
+    fireEvent.change(textarea, { target: { value: 'a'.repeat(300) } });
+    expect(screen.getByText('300')).toBeInTheDocument();
+
+    const searchError = getByTestId('DesignSystem-InlineMessage--Icon');
+    expect(searchError).toBeInTheDocument();
+  });
+});
```

Comment:
Please write a testcase for both Textarea and Input to check the error state when default value passed to Textfield is greater than the maxLimit